src/components/dapp-staking/modals/ModalRegisterDapp.vue

```diff
@@ -76,19 +78,30 @@ export default defineComponent({
     const stepsCount = 4;
     const registerForm = ref();
     const stepper = ref();
+    const isH160 = computed(() => store.getters['general/isH160Formatted']);
     const substrateAccounts = computed(() => store.getters['general/substrateAccounts']);
+    const { currentAccount } = useAccount();
+    const { callRegister } = useStakingH160(currentAccount);
+    const currentNetworkIdx = computed(() => store.getters['general/networkIdx']);
 
     const registerDapp = async (step: number): Promise<void> => {
       registerForm?.value?.validate().then(async (success: boolean) => {
         if (success) {
           if (step === stepsCount) {
-            const senderAddress = store.getters['general/selectedAddress'];
-            const result = await store.dispatch('dapps/registerDapp', {
-              dapp: data,
-              api: $api?.value,
-              senderAddress,
-              substrateAccounts: substrateAccounts.value,
-            } as RegisterParameters);
+            let result;
+            if (!isH160.value) {
+              const senderAddress = store.getters['general/selectedAddress'];
+              result = await store.dispatch('dapps/registerDapp', {
+                dapp: data,
+                api: $api?.value,
+                senderAddress,
+                substrateAccounts: substrateAccounts.value,
+              } as RegisterParameters);
+            } else {
+              if (currentNetworkIdx.value === endpointKey.SHIBUYA) {
```

Comment:
See my comment above. Information if network supports eth staking shouldn't be hardcoded.